src/geode/inspector/topology/private/brep_blocks_topology_impl.cpp

```diff
@@ -51,9 +54,109 @@ namespace geode
         bool BRepBlocksTopologyImpl::brep_vertex_blocks_topology_is_valid(
             index_t unique_vertex_index ) const
         {
-            const auto block_uuids =
-                components_uuids( brep_.component_mesh_vertices(
-                    unique_vertex_index, Block3D::component_type_static() ) );
+            const auto block_cmvs = brep_.component_mesh_vertices(
+                unique_vertex_index, Block3D::component_type_static() );
+            const auto block_uuids = components_uuids( block_cmvs );
+            if( block_uuids.size() == 1 )
+            {
+                const auto line_cmvs = brep_.component_mesh_vertices(
+                    unique_vertex_index, Line3D::component_type_static() );
+                const auto surface_cmvs = brep_.component_mesh_vertices(
+                    unique_vertex_index, Surface3D::component_type_static() );
+                absl::flat_hash_set< uuid > internal_surfaces;
+                for( const auto& surface_cmv : surface_cmvs )
+                {
+                    if( brep_.is_internal(
+                            brep_.surface( surface_cmv.component_id.id() ),
+                            brep_.block( block_uuids.front() ) ) )
+                    {
+                        if( !internal_surfaces
+                                 .emplace( surface_cmv.component_id.id() )
+                                 .second
+                            && line_cmvs.empty() )
+                        {
+                            if( verbose_ )
+                            {
+                                Logger::info( "Unique vertex with index ",
+                                    unique_vertex_index,
+                                    " is part of one block (",
+                                    block_uuids.front().string(),
+                                    "), and an internal surface (",
+                                    surface_cmv.component_id.id().string(),
+                                    ") but no line, with several component "
+                                    "mesh vertices along the surface." );
```

Comment:
This shouldn't be a problem, the problem is if there is a line boundary only to this surface and several cmvs on the same surface